backends/dpdk/control-plane/bfruntime_arch_handler.h

```diff
@@ -147,7 +147,7 @@ class BFRuntimeArchHandler : public P4RuntimeArchHandlerCommon<arch> {
         }
         auto *externType = p4info->add_externs();
         externType->set_extern_type_id(static_cast<p4rt_id_t>(typeId));
-        externType->set_extern_type_name(typeName);
+        externType->set_extern_type_name(typeName.string_view());
```

Comment:
Why is this required here?

Reply:
As I mentioned, Protobuf strings have two constructors: from `const char*` and `string_view`. In order to use `cstring` directly we need to make this conversion explicit.

Reply:
And I assume the reason is that we also have an implicit conversion to `const char*`? Is the follow-up PR working on that? Maybe it should be first.

Reply:
Right. And this is misused in many places, e.g. passing `cstring's` to C functions like `strlen`, improper checks for nulls, etc. :)

Reply:
The changes paired with the subsequent Pr look much better to me. Definitely an improvement.

Reply:
I might missed few cases here and there. However, I believe that we do not need lambdas in the example like in  https://github.com/p4lang/p4c/pull/4951#discussion_r1807919236 anymore.